packages/coding-style/src/Rector/ClassMethod/AnnotateThrowablesRector.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\ClassMethod;
+
+use Nette\Utils\Strings;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Throw_;
+use PhpParser\Node\Stmt\Use_;
+use PhpParser\Node\Stmt\UseUse;
+use Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwareGenericTagValueNode;
+use Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwarePhpDocTagNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use RuntimeException;
+
+/**
+ * Adds "throws" DocBlock to methods.
+ */
+final class AnnotateThrowablesRector extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Throw_::class];
+    }
+
+    /**
+     * From this method documentation is generated.
+     */
+    public function getDefinition(): RectorDefinition
+    {
+        return new RectorDefinition(
+            'Adds @throws DocBlock comments to methods that thrwo \Throwables.', [
+                new CodeSample(
+                     // code before
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                    ,
+                    // code after
+                    <<<'PHP'
+class RootExceptionInMethodWithDocblock
+{
+    /**
+     * This is a comment.
+     *
+     * @param int $code
+     * @throws \RuntimeException
+     */
+    public function throwException(int $code)
+    {
+        throw new \RuntimeException('', $code);
+    }
+}
+PHP
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @param Throw_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        $this->annotateMethod($node);
+
+        return $node;
+    }
+
+    private function isThrowableAnnotated(Throw_ $node): bool
+    {
+        $method = $node->getAttribute(AttributeKey::METHOD_NODE);
+
+        if ($method === null) {
+            throw new RuntimeException('This should not happen and is probably a bug. Please report it.');
```

Comment:
There is `ShouldNotHappenException` for this

Reply:
Also, throw can be used in a function as well. There is no `METHOD_NODE`.
It would make more sense to `return null;` instead

Reply:
I didn't took functions into account at all.

Do you think we should first release and then add support for functions or should have I to work first on implementing functions support?